File: cmd/cmd.pl
#!/usr/bin/env perl

use strict;
use Getopt::Long;
use Pod::Usage;

my $help;
my $message = '';
my $verbose = 0;

@ARGV and GetOptions(
    "h" => \$help,
    "m:s" => \$message,
    "v" => \$verbose,
    # TODO: add '-v' option for verbose mode
) or pod2usage(1);
pod2usage(-verbose => 2, -exitval => 2) if $help;

my $count = 0;

foreach $a(@ARGV) {
    # TODO: add '-v' option for verbose mode
    # and print each argument
    $count++;
    if($verbose != 0) {
        print $a . "\n";
    }

}

if($message ne "") {
    print($message . "\n");
}

print($count . "\n");

__END__

=head1 NAME

    printargs.pl - Print the number of arguments.

=head1 SYNOPSIS

    printargs.pl [options] [arguments]

=head1 OPTIONS

    -h      Show help message.

    -m MSG  Specify a custom message.


=head1 EXAMPLE

    printargs.sh a b c

    printargs.sh -m 'Arguments count: ' a b c

    printargs.sh -h

File: cmd/cmd.py
#!/usr/bin/env python

import argparse

parser = argparse.ArgumentParser(description='Print the number of arguments.')
parser.add_argument('arguments', metavar='ARG', type=str, nargs='*', help='some arguments')
parser.add_argument('-m', dest='message', default='', help='custom message')
parser.add_argument('-v', nargs='*')
# TODO: add '-v' option for verbose mode

args = parser.parse_args()

count = 0
argstr = ''

for a in args.arguments:
    # TODO: add '-v' option for verbose mode
    # and print each argument
    count += 1
    argstr += a + ' '

print(args)
print(args.arguments)

if args.message != '':
    print(args.message)

print(count)
if argstr != '':
	print('Arguments: ' + argstr)
